models/models.go

```diff
@@ -685,6 +721,21 @@ func (m *RouteManager) Destroy(guid string) error {
 	default:
 		return certErr
 	}
+	// Even though there is no gorm.Model for an `operations` table, it exists in the database.
+	// Attempting to delete a route record without deleting the associated operations first
+	// will cause a foreign-key constraint error, so we are manually deleting any associated
+	// operation records before attempting the route deletion.
+	hasActiveOperationsPending, err := m.hasActiveOperations(guid)
+	if err != nil {
+		return err
+	}
+	if !hasActiveOperationsPending {
+		if err := m.deleteAssociatedOperationsForRoute(guid); err != nil {
+			return err
+		}
+	} else {
+		return errors.New("cannot delete route: there are active operations at this time. Please try again later")
```

Comment:
I'm a little torn on this - deleting the operations and route separately means it's _possible_ that we'll delete the operations, then in the intervening time a new operation will be created. 
A `delete ... cascade` would side-step this, but it's a lot less clear what we're doing.

Given the low churn on this, probably either one is OK 🤷🏻‍♂️